src/Sys/InspectSeoHandler.php

```diff
@@ -0,0 +1,177 @@
+        private readonly Flags           $flags,
+        private readonly Pbjx            $pbjx,
+        private readonly LoggerInterface $logger = new NullLogger(),
+    )
+    {
+    }
+
+    public function handleCommand(Message $command, Pbjx $pbjx): void
+    {
+        $searchEngines = $command->get('search_engines', ['google']);
+        $retryCommand = clone $command;
+        $retryCommand->clear('search_engines');
+        $nodeRef = $retryCommand->get('node_ref');
+
+        try {
+            $node = $this->ncr->getNode($nodeRef);
+        } catch (NodeNotFound $e) {
+            $this->logger->error('Unable to get node for search engines processing.', [
+                'exception' => $e,
+                'node_ref' => $retryCommand->get('node_ref')
+            ]);
+            return;
+        }
+
+        foreach ($searchEngines as $searchEngine) {
+            $methodName = 'checkIndexStatusFor' . ucfirst($searchEngine);
+
+            if (!method_exists($this, $methodName)) {
+                continue;
+            }
+
+            $retryCommand = $this->$methodName($command, $node);
+        }
+
+        if (empty($retryCommand->get('search_engines'))) {
+            return;
+        }
+
+        $retries = $command->get('ctx_retries');
+        $maxRetries = $this->flags->getInt('inspect_seo_max_retries', 5);
+
+        if ($retries < $maxRetries) {
+            $retryCommand->set('ctx_retries', $retryCommand->get('ctx_retries') + 1);
+
+            $pbjx->sendAt(
+                $retryCommand,
+                strtotime("+" . $this->flags->getInt('inspect_seo_retry_delay', 15) . "minutes"),
+                "{$nodeRef}.inspect-seo"
+            );
+        }
+    }
+
+
+    public function checkIndexStatusForGoogle(Message $command, Message $node): Message
+    {
+        $retryCommand = clone $command;
+        $request = new \Google_Service_SearchConsole_InspectUrlIndexRequest();
+        $request->setSiteUrl($this->flags->getString('inspect_seo_google_site_url'));
+        $url = UriTemplateService::expand(
+            "{$node::schema()->getQName()}.canonical", $node->getUriTemplateVars()
+        );
+        $request->setInspectionUrl($url);
+        $client = new \Google_Client();
+        $client->addScope(\Google_Service_SearchConsole::WEBMASTERS_READONLY);
+
+        $response = null;
+
+        try {
+            $base64EncodedAuthConfig = Crypto::decrypt(getenv('GOOGLE_SEARCH_CONSOLE_API_SERVICE_ACCOUNT_OAUTH_CONFIG'), $this->key);
+            $client->setAuthConfig(json_decode(base64_decode($base64EncodedAuthConfig), true));
+            $service = new \Google_Service_SearchConsole($client);
+            $response = $service->urlInspection_index->inspect($request);
+        } catch (\Throwable $e) {
+            $this->logger->error('An error occurred in checkIndexStatus for google.', [
+                'exception' => $e,
+                'node_id' => $node->get('node_id'),
```

Comment:
nodeRef would be more helpful I think and to get the id is should be $node->get('_id')